packages/utils/src/normalize.ts

```diff
@@ -51,170 +53,173 @@ export function normalizeToSize<T>(
-  if (depth === 0) {
-    return serializeValue(value);
+  // if the value has a `toJSON` method, bail and let it do the work
+  const valueWithToJSON = value as unknown & { toJSON?: () => string };
+  if (valueWithToJSON && typeof valueWithToJSON.toJSON === 'function') {
+    try {
+      return valueWithToJSON.toJSON();
+    } catch (err) {
+      return `**non-serializable** (${err})`;
+    }
   }
 
-  // If value implements `toJSON` method, call it and return early
-  if (value !== null && value !== undefined && typeof value.toJSON === 'function') {
-    return value.toJSON();
+  // get the simple cases out of the way first
+  if (value === null || (['number', 'boolean', 'string'].includes(typeof value) && !isNaN(value))) {
+    return value as Primitive;
   }
 
-  // `makeSerializable` provides a string representation of certain non-serializable values. For all others, it's a
-  // pass-through. If what comes back is a primitive (either because it's been stringified or because it was primitive
-  // all along), we're done.
-  const serializable = makeSerializable(value, key);
-  if (isPrimitive(serializable)) {
-    return serializable;
+  const stringified = stringifyValue(key, value);
+
+  // Anything we could potentially dig into more (objects or arrays) will have come back as `"[object XXXX]"`.
+  // Everything else will have already been serialized, so if we don't see that pattern, we're done.
+  if (!stringified.startsWith('[object ')) {
+    return stringified;
   }
 
-  // Create source that we will use for the next iteration. It will either be an objectified error object (`Error` type
-  // with extracted key:value pairs) or the input itself.
-  const source = getWalkSource(value);
+  // we're also done if we've reached the max depth
+  if (depth === 0) {
+    // At this point we know `serialized` is a string of the form `"[object XXXX]"`. Clean it up so it's just `"[XXXX]"`.
+    return stringified.replace('object ', '');
+  }
 
-  // Create an accumulator that will act as a parent for all future itterations of that branch
+  // Create source that we will use for the next iteration. Because not all of the properties we care about on `Error`
+  // and `Event` instances are ennumerable, we first convert those to plain objects. (`convertToPlainObject` is a
+  // pass-through for everything else.)
+  const source = convertToPlainObject(value);
+
+  // Create an accumulator that will act as a parent for all future iterations of this branch, and keep track of the
+  // number of properties/entries we add to it
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const acc: { [key: string]: any } = Array.isArray(value) ? [] : {};
+  let numAdded = 0;
 
-  // If we already walked that branch, bail out, as it's circular reference
+  // If we've already visited this branch, bail out, as it's circular reference
   if (memoize(value)) {
     return '[Circular ~]';
   }
```

Comment:
(optional because out of scope) We could move this block upwards above the `convertToPlainObject` call, to bail out a tiny bit earlier if we have a circ-ref.

Reply:
Good call. Done.